src/irmin-pack/inode_intf.ml

```diff
@@ -83,6 +83,24 @@ module type INTER = sig
 
     val integrity_check : t -> bool
     (** Checks the integrity of an inode. *)
+
+    module Tree : sig
+      (** {1 Concrete trees} *)
+
+      type kind = Contents of metadata | Node [@@deriving irmin]
+      type entry = { name : step; kind : kind; hash : hash } [@@deriving irmin]
```

Comment:
Instead of `kind : kind; hash : hash` we could use a `kinded_hash` as introduced in `tree.`ml

Reply:
I was just trying to inline that definition to avoid an indirection in the JSON schema file.